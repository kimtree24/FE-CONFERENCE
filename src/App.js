// App.js
import React, { useState } from "react";
import axios from "axios";

  // 사용자 ID로 API에서 사용자 이름을 가져오는 함수
  export const fetchUserName = async (userId) => {
    try {
      const response = await axios.get(
        `https://jsonplaceholder.typicode.com/users/${userId}`
      );
      return response.data.name;
    } catch {
      throw new Error("사용자를 찾을 수 없습니다. ID를 확인하세요.");
    }
  };

  // 숫자 문자열을 받아 합을 계산하는 함수
  export const calculateSum = (numbersString) => {
    const numbersArray = numbersString.split(",").map((numInput) => {
      const num = Number(numInput)
      console.log(num)
      if (isNaN(num)){
        throw new Error("Nan 포함");
      }
      if (num < 0){
        throw new Error("음수 값 불가")
      }
      return num;
    });
    return numbersArray.reduce((acc, num) => acc + num, 0);
  };


function App() {
  const [id, setId] = useState("");
  const [numbers, setNumbers] = useState("");
  const [userName, setUserName] = useState("");
  const [sum, setSum] = useState(null);

  // 버튼 클릭 시 호출되는 메인 함수
  const handleFetchAndCalculate = async () => {
    try {
      // 사용자 이름 가져오기
      const name = await fetchUserName(id);
      setUserName(name);

      // 숫자 합계 계산하기
      const totalSum = calculateSum(numbers);
      setSum(totalSum);
    } catch (err) {
      console.log(err.message);
      setUserName("");
      setSum(null);
    }
  };

  return (
    <div style={{ textAlign: "center", marginTop: "50px" }}>
      <h1>사용자 정보와 숫자 합계 계산</h1>
      <div>
        <input
          type="text"
          value={id}
          onChange={(e) => setId(e.target.value)}
          placeholder="ID 값을 입력하세요"
          style={{ padding: "10px", fontSize: "16px", marginRight: "10px" }}
        />
        <input
          type="text"
          value={numbers}
          onChange={(e) => setNumbers(e.target.value)}
          placeholder="숫자들을 입력하세요 (예: 4,5,9)"
          style={{ padding: "10px", fontSize: "16px" }}
        />
        <button
          onClick={handleFetchAndCalculate}
          style={{ padding: "10px 20px", fontSize: "16px", marginLeft: "10px" }}
        >
          조회하기
        </button>
      </div>
      {userName && (
        <div style={{ marginTop: "20px", fontSize: "18px" }}>
          <p>
            <strong>사용자 이름:</strong> {userName}
          </p>
          <p>
            <strong>숫자들의 합:</strong> {sum}
          </p>
        </div>
      )}
    </div>
  );
}

export default App;