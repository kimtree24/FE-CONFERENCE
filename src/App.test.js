import { render, screen } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

describe(`<App />`, () => {
  it("renders component correctly", () => {
    render(<App />);
    expect(screen.getByTestId("p-tag")).toMatchSnapshot();
    expect(screen.getAllByTestId("p-tag")).toHaveLength(1);
    expect(screen.getAllByTestId("p-tag")[0]).toHaveTextContent(
      "Edit src/App.js and save to reload"
    );
  });
});