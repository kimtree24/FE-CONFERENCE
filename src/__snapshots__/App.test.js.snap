// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders component correctly 1`] = `
<p
  data-testid="p-tag"
>
  Edit 
  <code>
    src/App.js
  </code>
   and save to reload.
</p>
`;
